### DISCUSSION TITLE
Welcome to HippoPlayer Discussions!

### DISCUSSION THREAD

**Discussion by emoon** [emoon/HippoPlayer#171] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Kepalu** [comment]
Hey emoon, I really love this software! Potentially the best mod player for Linux. I really love how all the track's information is displayed right away (comments, instruments, samples) - it is extremely useful! And this alone makes it better than XMPlay in my opinion.

However there are many issues and lacking options. The biggest ones so far in my opinion are these 2:
1. Loading files. - there is no way of simply dragging the files into the player, there is only option to add new files into the playlist, but no way of replacing it with a new file, or new directory, etc.
2. The lack of playback bar that would allow you to skip to a specific moment of a track.

Thank you for you work, keep it up! Linux desperately needs a good mod player!

**Comment by emoon** [comment]
Hi!

First thanks for those kind words. Comments like that make it more motivating to develop the program :)

As you noted there are several features lacking.

1. It should be possible (with the latest version) to drag'n'drop files (it has to be dropped currently inside the playlist window) That being said there are lots of improvements needed for the playlist. There is a reason why hippo is still in 0.1.x :)
2. Yes, this is something I will work on next once the playback settings system is in place (almost done)

Thanks again :)

**Comment by Kepalu** [comment]
Hey, you're welcome! Any complements are well deserved ;-)

Sorry, I just realized that AUR I'm using is a bit outdated (v. 0.1.23-1):
https://aur.archlinux.org/packages/hippoplayer/
I'm a very vanilla user, so I'll just have to wait for the update and check out the file-dragging option :)

Not sure if this has changed in the newest version, but I have observed another important issue: whatever customizations you do in the software (like changing the size of the columns in the playlist) they don't get saved after exiting the program, and after launching it again it has the default settings.

Aside from that let me express this: I've been looking for a native Linux MOD player for ages. I refuse to even install WINE, as it comes with lots of risks, not to mention that using it is painful. I know that there are many people like me who maybe aren't expressing this need online too often. All the MOD players native to Linux that I came across in the past were terrible. Either the quality of reproducing the sound was off, or there was no easy access to the track's information. Like I have mentioned in my first post: the right side of the HippoPlayer displaying all the information in one column is something I was dreaming of for ages, and couldn't get that even on Windows players. The other thing that makes this software really useful is breaking the MOD file into as many tracks as it contains (that are otherwise hidden in most of the players).

![HippoPlayer](https://user-images.githubusercontent.com/48812028/105619658-c7549f00-5da9-11eb-9f16-68cf271e92ea.png)

I love it! :D

**Comment by emoon** [comment]
Cool. I think the AUR is outdated as it has some problems building the code. Basic drag'n'drop support was added in `0.1.25` I didn't do that one myself and I think someone reported it.

Yeah. Settings is something I working on for the next (0.2) version that is coming. The main feature I have been working on there is to have settings for the playback like this 

![image](https://user-images.githubusercontent.com/350822/105812310-37cc0d80-5fae-11eb-8277-30d42c258c94.png) 

some plugins (such as openmpt) will allow you to tweak all the settings in real-time (which isn't that common to what I have seen with other players)

Re Linux players: Yes, In my case my main platform is Linux these days, when I started on Hippo I was mainly using mac which suffers from the same problem so a win for both platforms I say :)

Re sub-songs: Yeah, this is something I found annoying in all other players they would just show `Song name (4/5)` That way could could never say "I just want to have one sub-song in my playlist). The plan later on is to support a (optional) tree structure here so it would look like

```
 [-] Hamster Ball
    Name 1
    Name 2
```

**Reply by Kepalu** [reply]
All this sounds great! I simply can not wait for the newer versions! :) 
Thank you for your hard work and good luck!

**Comment by illettante** [comment]
Just here to give some love. Long time Windows and XMPlay user finally dragooned onto macos by my workplace. Tried XMPlay on WINE, Colibri, Audio Overload, QuodLibet, RetroPlayer, CMUS, MOC, MPD, Cog (Kode54 fork), DeaDBeeF, and now HippoPlayer. I've got a 128gb thumb drive with 20k+ tracker music files scraped off a bunch of places like HornetArchive, ModArchive, AmiNet, HVSC etc and it loads the whole lot very quickly, doesn't crash itself attempting to verify them, and so far hasn't choked to death on anything, so it's currently leading the pack as the best tracker module player on macos as voted by me. Bravo.

**Reply by emoon** [reply]
@illettante Thanks a lot for the praise! :) 

As I wrote in the earlier issue I'm right now in the middle of a big rewrite and expect something even better when I get that up and running!
